tests/sentry/sentry_metrics/test_multiprocess_steps.py

```diff
@@ -287,6 +294,49 @@ def test_process_messages(mock_indexer) -> None:
     compare_message_batches_ignoring_metadata(new_batch, expected_new_batch)
 
 
+@patch("sentry.sentry_metrics.multiprocess.get_indexer", return_value=MockIndexer())
+def test_transform_step(mock_indexer) -> None:
+    config = get_ingest_config(UseCaseKey.RELEASE_HEALTH)
+
+    message_payloads = [counter_payload, distribution_payload, set_payload]
+
+    produce_step = Mock()
+    transform_step = TransformStep(next_step=produce_step, config=config)
```

Comment:
I am still not 100% sure why this was failing before and the other tests did not catch it.

Reply:
Before this PR, there was no test that actually invoked the `TransformStep`, which sort of makes sense because it's a very thin wrapper around `process_messages` which is easier to call directly.

Reply:
Don't we have an end to end test that tests the whole ProcessingStrategy ?